specification/dimensions/commitmentdiscountid.md

```diff
@@ -0,0 +1,32 @@
+# Commitment Discount ID
+
+A commitment-based discount is a contractual commitment for an amount of usage or spend, for a specified term, in exchange for a discounted unit price. Commitment-based discounts can cover either usage or cost of various resources such as virtual machines or databases. A provider can offer various pricing models, payment options, and term durations for commitment-based discounts.
```

Comment:
```suggestion
A commitment-based discount is a contractual commitment for a minimum amount of usage or spend throughout a specified term, in exchange for discounted unit pricing on that amount. The commitment may be based on quantities of resource units or monetary value, with various payment options and timeframes.
```

- Additional clarity on discount (can be many unit prices, and only for the amount committed)
- Removed reference to specific resources in case providers do not or stop offering commitments on those resources.

Reply:
I would prefer if we didn't have to say 'minimum amount' to make it more general- might be true today cloud commitments but if we think about the broader context, I could see someone saying you had to average over a period of time (e.g. for people, average 40 hours/week of a consultants time over some period)

Reply:
Rather than minimum, it’s spend committed for a time period (not minimum spend).  I liked the original language before minimum was added.

Reply:
Something like:

> A commitment-based discount is a contractual commitment for an amount of usage or spend committed for a time period, in exchange for discounted unit pricing on that amount. The commitment may be based on quantities of resource units or monetary value, with various payment options and timeframes.

?

Reply:
Updated the suggestion to remove the "minimum", to mitigate marginal changes to the spec if providers add additional terms to their definition. 

There is an important distinction to emphasize when removing "minimum", that the `term` of the entire commitment is not necessarily the `term` of the committed usage or spend amount. Hopefully "throughout" makes that clear.

In @udam-f2 's scenario, if applied to compute, then there is no distinction for commit term and CBD term : average of $40/hr Compute over a 1-year period (trued-up at year end). Rather than the current commit for an absolute $40/hr Compute over a 1-year period (enforceable per hour).

Reply:
By the way, should it be `time frame` instead of `timeframe`?